host-controller/src/main/java/org/jboss/as/host/controller/model/jvm/JvmOptionsBuilderFactory.java

```diff
@@ -167,6 +168,19 @@ boolean checkOption(boolean condition, String jvm, String option, String schemaE
     }
 
     private static class SunJvmOptionsBuilder extends JvmOptionsBuilder {
+        public static final int JVM_MAJOR_VERSION;
+
+        static {
+            int vmVersion;
+            try {
+                String vmVersionStr = WildFlySecurityManager.getPropertyPrivileged("java.specification.version", null);
+                vmVersion = Integer.valueOf(vmVersionStr.substring(2));
```

Comment:
Yes it is an assumption that might break in a future.
I just copied existing code :) will add some more safeguards.